### PR TITLE
add fare adjustments in r5

### LINKED ISSUE #892: add fare adjustment in r5

### PR DESCRIPTION
**Pull request by zishanbilal**
Fixed #892

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/949)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/choice/mode/ModeSubsidy.scala [hunk 1]
```diff
@@ -56,9 +62,11 @@ object ModeSubsidy {
 
   def test(): Unit = {
     val ms = new ModeSubsidy(loadSubsidies("test/input/beamville/subsidies.csv"))
-    assert(ms.getSubsidy(BeamMode.RIDE_HAIL, Some(5), Some(30000)) == 4)
-    assert(ms.getSubsidy(BeamMode.RIDE_HAIL, Some(25), Some(30000)) == 3)
-    assert(ms.getSubsidy(BeamMode.RIDE_HAIL, None, None) == 0)
+    assert(ms.getSubsidy(BeamMode.RIDE_HAIL, Some(5), Some(30000), None, None).getOrElse(0) == 4)
```
Comment (wrashid) [inline src/main/scala/beam/agentsim/agents/choice/mode/ModeSubsidy.scala]: move to tests

File: src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala [hunk 2]
```diff
@@ -110,11 +110,26 @@ class ModeChoiceMultinomialLogit(val beamServices: BeamServices, val model: Mult
         case _ =>
           altAndIdx._1.costEstimate
       }
-      val subsidy = beamServices.modeSubsidies.getSubsidy(
-        mode,
-        attributesOfIndividual.age,
-        attributesOfIndividual.income.map(x => x.toInt)
-      )
+
```
Comment (wrashid) [inline src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala]: refactor: code duplication with PersonAgent

**Comment by zishanbilal** [comment]
test!

**Comment by zishanbilal** [comment]
test!

**Pull request merged**